zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/UserTaskHandler.java

```diff
@@ -249,4 +210,75 @@ private void createTaskEntity(final TaskEntity entity, final Record<UserTaskReco
           .ifPresent(c -> entity.setFormId(c.formId()).setFormVersion(c.formVersion()));
     }
   }
+
+  private void handleAssignment(final Record<UserTaskRecordValue> record, final TaskEntity entity) {
+    entity.getChangedAttributes().add("assignee");
+    if (ExporterUtil.isEmpty(record.getValue().getAssignee())) {
+      entity.setAssignee(null);
+    } else {
+      entity.setAssignee(record.getValue().getAssignee());
+    }
+
+    updateChangedAttributes(record, entity);
+  }
+
+  /**
+   * Applies changes to the user task fields based on the attributes in the {@link
+   * UserTaskRecordValue}.
+   *
+   * <p>This method can be used for updating fields either:
+   *
+   * <ul>
+   *   <li>As a result of user task corrections configured while completing task listener jobs.
+   *   <li>As a result of regular user task updates.
+   * </ul>
+   *
+   * @param record the record containing user task data and changed attributes
+   * @param entity the task entity to be updated
+   */
+  private void updateChangedAttributes(
+      final Record<UserTaskRecordValue> record, final TaskEntity entity) {
+    final var value = record.getValue();
+
+    for (final String attribute : value.getChangedAttributes()) {
+      entity.getChangedAttributes().add(attribute);
+
+      switch (attribute) {
+        case "assignee" -> entity.setAssignee(value.getAssignee());
+        case "candidateGroupsList" ->
+            entity.setCandidateGroups(value.getCandidateGroupsList().toArray(new String[0]));
+        case "candidateUsersList" ->
+            entity.setCandidateUsers(value.getCandidateUsersList().toArray(new String[0]));
+        case "dueDate" -> entity.setDueDate(ExporterUtil.toOffsetDateTime(value.getDueDate()));
+        case "followUpDate" ->
+            entity.setFollowUpDate(ExporterUtil.toOffsetDateTime(value.getFollowUpDate()));
+        case "priority" -> entity.setPriority(value.getPriority());
```

Comment:
💭 We could use `UserTaskRecord.ASSIGNEE` and other constants here if we made `zeebe-protocol-impl` a compile-scope dependency. However, since `zeebe-protocol-impl` was not explicitly defined as a production dependency in the `camunda-exporter` pom file before, I opted to retain string literals as used previously.

❔I’d appreciate your thoughts on introducing `zeebe-protocol-impl` as a compile-scope dependency to improve the maintainability or if there are reasons to avoid making it a compile dependency.
@panagiotisgts @korthout @ana-vinogradova-camunda

Reply:
Personally, I'd avoid it, as to not open the door for other unintended usages. But the owning team may have different ideas.

Reply:
💬 Thanks, Nico! I agree with your point about avoiding `zeebe-protocol-impl` as a compile-scope dependency to prevent unintended usages. Given this, I've decided to stick with string literals for now to keep things simple and avoid introducing new dependencies in the production code.

If the owning team has a different preference or if these attribute constants become part of the `zeebe-protocol` itself in the future, we can revisit this decision.